transcom/mymove
# [MB-1388] Imported 2020 400ng data; fixed some migrator issues

## Description

This PR imports the [2020 400ng rates](https://docs.google.com/spreadsheets/d/1OFA-3jfHFfRy7RbamGOdT0bzojY7RQnmrl3JaxT3CR0/edit#gid=1070545589) and related data via a migration.  In addition, it addresses a few issues in `pkg/migrate` that were causing a SQL migration to fail under our migration system but work fine when sent to `psql`.

I also updated our documentation for doing a 400ng import.  Because we are trying to avoid having Postgres generate our UUIDs (and getting a different UUID for these records in each environment), I had to change the process some.

## Reviewer Notes

Most of the data import follows the same [general process](https://github.com/transcom/mymove/blob/master/docs/data/tariff400ng-yearly-import.md) as before.  I just pushed the data into a local database instead, then used `pg_dump` to extract it to SQL (using the fast `COPY` method) that became the included migration.

I did audit the `tariff400ng_zip3s` table as well since zip3 and service area data are contained in the spreadsheet but not fully imported like the other imported tables (zip3/zip5 data is not scoped to a time period currently).  I did find one needed change ... zip `629` (Vienna, IL) had a rate area of `252` but should have been `256`.  There's a separate migration to fix that.

The two issues I addressed in the migrator are:
- When parsing `COPY` statements, the parser was not matching the `tariff400ng_*` tables because the regex pattern didn't allow numbers in table names.
- Previously, an assumption was made that only one `COPY` would appear in any migration.  Since a 400ng update involves a half dozen or so tables and we're now using `pg_dump` (and `COPY`) to create the needed migration, that assumption was no longer valid.  I also didn't want to split up the tables across separate migrations because I'd like all the data imported in a single transaction.

## Setup

To test the import:
- Make sure you rebuild your `milmove` binary: `rm bin/milmove`, then `make bin/milmove`
- `make db_dev_reset db_dev_migrate`
- Spot-check the data imported into the following tables against the 2020 400ng spreadsheet:
  - `tariff400ng_full_pack_rates` -- see “Additional Rates” tab, lines 15-30
  - `tariff400ng_full_unpack_rates` -- see “Additional Rates” tab, lines 15-30 — see red text (unpack doesn't depend on weight)
  - `tariff400ng_item_rates` -- see "Additional Rates" and "Accessorials" tabs
  - `tariff400ng_linehaul_rates` -- see "Linehaul" tab
  - `tariff400ng_service_areas` -- see "Geographical Schedule" tab
  - `tariff400ng_shorthaul_rates` -- see "Additional Rates" tab, lines 65-70
- Also check for the changes noted above for zip `629` in the `tariff400ng_zip3s` table.  Full list of zip3s-to-service-areas is on the "Base Point City" tab
- Some additional spot checking queries can be found in the [new docs](https://github.com/transcom/mymove/blob/rr-MB-1388-import_2020_400ng/docs/data/tariff400ng-yearly-import.md#spot-check-for-correct-data)

To run server tests for the `pkg/migrate` fixes:
- `make server_test`

## Code Review Verification Steps

* Any new migrations/schema changes:

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-1388) for this change
* [2020 400NG Baseline Rates](https://docs.google.com/spreadsheets/d/1OFA-3jfHFfRy7RbamGOdT0bzojY7RQnmrl3JaxT3CR0/edit#gid=1070545589) spreadsheet that serves as the basis for this import
* [Existing import process documentation](https://github.com/transcom/mymove/blob/master/docs/data/tariff400ng-yearly-import.md) (updated in this PR)
